client/internal/mockserver.go

```diff
@@ -112,6 +112,11 @@ func (m *MockServer) EnableExpectMode() {
 
 func (m *MockServer) handlePlainHttp(w http.ResponseWriter, r *http.Request) {
 	msgBytes, err := io.ReadAll(r.Body)
+	if err != nil {
+		// request could not be read for some reason
+		http.Error(w, err.Error(), 500)
```

Comment:
We don't wan't to accidentally leak sensitive info that may be in the error message, so best not to use it directly in the response.
```suggestion
		http.Error(w, "Error reading request body", http.StatusInternalServerError)
```

Reply:
I think in this case it is OK, since it's a mock server we only expect to use in test?

Reply:
Yes, you are right. I missed that it is a mock server.

Reply:
Can still use `http.StatusInternalServerError`, no harm in that.